### PR TITLE
+act #19704 Fix UnboundedControlAwareMessageQueueSemantics verification

### PR DESCRIPTION
**Pull request by virsox**
I hope I've followed all the contributions guidelines correctly.

### CODE REVIEW

**Comment by drewhk** [comment]
OK TO TEST

**Comment by drewhk** [comment]
refs #19704

**Comment by drewhk** [comment]
I'm afraid this won't be binary compatible though.

**Comment by virsox** [comment]
Sorry, any chance of this being cause by another PR? It seems https://github.com/akka/akka/pull/19564 added some new tests that are failing, but I couldn't connect with this PR

**Comment by johanandren** [comment]
Yes, the failure was unrelated to this PR

**Comment by johanandren** [comment]
PLS BUILD

File: akka-actor/src/main/scala/akka/dispatch/Mailbox.scala [hunk 1]
```diff
@@ -842,7 +842,7 @@ final case class UnboundedControlAwareMailbox() extends MailboxType with Produce
 }
 
 object UnboundedControlAwareMailbox {
-  class MessageQueue extends ControlAwareMessageQueueSemantics with UnboundedMessageQueueSemantics {
+  class MessageQueue extends UnboundedControlAwareMessageQueueSemantics with java.io.Serializable {
```
Comment (ktoso) [inline akka-actor/src/main/scala/akka/dispatch/Mailbox.scala]: Why making it extend Serializable?

File: akka-actor/src/main/scala/akka/dispatch/Mailbox.scala [hunk 2]
```diff
@@ -862,7 +862,7 @@ final case class BoundedControlAwareMailbox(capacity: Int, override final val pu
 }
 
 object BoundedControlAwareMailbox {
-  class MessageQueue(val capacity: Int, val pushTimeOut: FiniteDuration) extends BoundedControlAwareMessageQueueSemantics {
+  class MessageQueue(val capacity: Int, val pushTimeOut: FiniteDuration) extends BoundedControlAwareMessageQueueSemantics with java.io.Serializable {
```
Comment (viktorklang) [inline akka-actor/src/main/scala/akka/dispatch/Mailbox.scala]: I assume @ktoso has the same Serializable question here

**Comment by rkuhn** [comment]
It seems to be binary compatible, but I, too, wonder about the Serializable extension.

**Comment by drewhk** [comment]
Isn't it only reported as compatible as these things are wired dynamically?

**Comment by rkuhn** [comment]
Which method call do you expect to fail with a LinkageError after these changes?

**Comment by drewhk** [comment]
One class (MessageQueue) now extends a different superclass. Isn't that a problem? I don't know which method since I have not much recollection of the mailbox selection API anymore.

**Comment by rkuhn** [comment]
The additional type that was inserted into the hierarchy does not add any methods, all type tests will still come back with the same result apart from adding the one `true` that was missing. If there was a method change that causes binary incompatibility then MiMa would flag it.

**Comment by drewhk** [comment]
ok

File: akka-actor/src/main/scala/akka/dispatch/Mailbox.scala (hunk 2 above)
Comment (virsox) [inline akka-actor/src/main/scala/akka/dispatch/Mailbox.scala]: The tests send `MessageQueue` as messages, so they fail if the `MessageQueue` is not Serializable. It felt odd to me, but as far as I could track all other `MessageQueues` are also Serializable (by inheritance, e.g. `BoundedDequeBasedMailbox.MessageQueue` extends `LinkedBlockingDeque`, which is Serializable). Also, it makes sense if we consider queues store Messages, which are Serializable as well.

Therefore, for consistency sake, I decided to make `BoundedControlAwareMailbox.MessageQueue` Serializable. However, I must say that I haven't thought about binary compatibility here. If needed, I can remove the extends and create some other tests for this functionality

Comment (rkuhn) [inline akka-actor/src/main/scala/akka/dispatch/Mailbox.scala]: Hmm, interesting. I guess this is at least consistent then.

**Comment by drewhk** [comment]
So this is basically LGTM. Any objections to merge it?

**Comment by rkuhn** [comment]
LGTM, thanks @virsox !

**Pull request merged**